plugins/logging/pkg/backend/sync.go

```diff
@@ -0,0 +1,235 @@
+package backend
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/gogo/protobuf/proto"
+	opnicorev1beta1 "github.com/rancher/opni/apis/core/v1beta1"
+	loggingv1beta1 "github.com/rancher/opni/apis/logging/v1beta1"
+	capabilityv1 "github.com/rancher/opni/pkg/apis/capability/v1"
+	opnicorev1 "github.com/rancher/opni/pkg/apis/core/v1"
+	"github.com/rancher/opni/pkg/auth/cluster"
+	"github.com/rancher/opni/pkg/capabilities/wellknown"
+	"github.com/rancher/opni/pkg/resources"
+	"github.com/rancher/opni/pkg/util"
+	"github.com/rancher/opni/plugins/logging/pkg/apis/node"
+	loggingerrors "github.com/rancher/opni/plugins/logging/pkg/errors"
+	"go.uber.org/zap"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/types/known/timestamppb"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/retry"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
```

Comment:
The LoggingBackend should be relatively generic - this is the intersection between the generic API (CapabilityBackend) and the implementation-specific API (NodeLoggingCapability). Both servers should for the most part forward requests to an implementation-specific "ClusterDriver" (or similar).